libraries/node-core-library/src/FileSystem.ts

```diff
@@ -729,22 +818,92 @@ export class FileSystem {
   }
 
   /**
-   * Copies a file from one location to another.
+   * Copies a single file from one location to another.
    * By default, destinationPath is overwritten if it already exists.
-   * Behind the scenes it uses `fs.copyFileSync()`.
+   *
+   * @remarks
+   * The `copyFile()` API cannot be used to copy folders.  It copies at most one file.
+   * Use {@link FileSystem.copyFiles} if you need to recursively copy a tree of folders.
+   *
+   * The implementation is based on `copySync()` from the `fs-extra` package.
    */
   public static copyFile(options: IFileSystemCopyFileOptions): void {
+    const existsBehavior: AlreadyExistsBehavior = options.alreadyExistsBehavior !== undefined
+      ? options.alreadyExistsBehavior : AlreadyExistsBehavior.Overwrite;
+
+    if (FileSystem.getStatistics(options.sourcePath).isDirectory()) {
```

Comment:
Can you just try to do the copy and throw this if the error says it's a directory?

Reply:
@iclanton @sachinjoseph I don't think so. The API will actually do a recursive copy of the directory and its contents.  Algorithmically `fx-extra` is correct to model `copySync()` as a thing that can copy files or folders.

But ergonomically, I would find it highly counterintuitive to write this:
```
copyFile(fromPath, toPath);
```
...and have 1,000 files get copied. This is why we have XCOPY and why the Unix `cp` command refuses to copy directories unless you explicitly specify `--recursive`.